From b2bf30165f2f69709cf4817bbce55c29688cf2fd Mon Sep 17 00:00:00 2001
From: Olivier Fourdan <ofourdan@redhat.com>
Date: Mon, 13 Jun 2016 14:00:47 +0200
Subject: [PATCH 1/7] wayland: Mark pending moved as moved

The result flag needs to be marked as moved even for pending moves,
otherwise the window's unconstrained_rect doesn't get updated in
meta_window_move_resize_internal() and the anchor grab is wrong.

https://bugzilla.gnome.org/show_bug.cgi?id=764180
---
 src/wayland/meta-window-wayland.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/wayland/meta-window-wayland.c b/src/wayland/meta-window-wayland.c
index b360f91..5181829 100644
--- a/src/wayland/meta-window-wayland.c
+++ b/src/wayland/meta-window-wayland.c
@@ -310,6 +310,7 @@ meta_window_wayland_move_resize_internal (MetaWindow                *window,
 
       if (new_x != window->rect.x || new_y != window->rect.y)
         {
+          *result |= META_MOVE_RESIZE_RESULT_MOVED;
           wl_window->has_pending_move = TRUE;
           wl_window->pending_move_x = new_x;
           wl_window->pending_move_y = new_y;
-- 
2.8.1

